lib/api/core/plugins/pluginsManager.js

```diff
@@ -473,295 +636,269 @@ function initControllers (controllers, routes, plugin, pluginName) {
+        if (!pluginsManager.controllers[`${plugin.name}/${route.controller}`][route.action]) {
           // eslint-disable-next-line no-console
-          console.error(`[Plugin Manager] Error initializing route ${route.url}: undefined action ${route.action}`);
+          console.warn(clc.yellow(`${errorRoutePrefix} Undefined action "${clc.bold(route.action)}"`));
           return false;
         }
 
 
         if (allowedVerbs.indexOf(route.verb.toLowerCase()) === -1) {
           // eslint-disable-next-line no-console
-          console.error(`[Plugin Manager] Error initializing route ${route.url}: only following http verbs are allowed: ${allowedVerbs.join(', ')}`);
+          console.warn(clc.yellow(`${errorRoutePrefix} Only following http verbs are allowed: "${clc.bold(allowedVerbs.join(', '))}"`));
           return false;
         }
 
-        route.url = '/' + pluginName + route.url;
-        route.controller = pluginName + '/' + route.controller;
+        route.url = '/' + plugin.name + route.url;
+        route.controller = plugin.name + '/' + route.controller;
 
-        debug('[%s] binding HTTP route "%s" to controller "%s"', pluginName, route.url, route.controller);
-        routes.push(route);
+        debug('[%s] binding HTTP route "%s" to controller "%s"', plugin.name, route.url, route.controller);
+        pluginsManager.routes.push(route);
+
+        return true;
       }
     });
   }
-}
-
-/**
- * Emit event
- *
- * @param {EventEmitter} emitter
- * @param {string} event
- * @param {*} data
- * @returns {Promise}
- */
-function triggerHooks(emitter, event, data) {
-  emitter.emit(event, data);
 
-  return Bluebird.resolve(data);
+  return true;
 }
 
 /**
- * Chain call all attached functions plugins on the specific event
- *
- * @param {object} pipes
- * @param {string} event
- * @param {*} data
- * @returns {Promise}
+ * @param {PluginsManager} pluginsManager
+ * @param {object} plugin
+ * @return {boolean} return true if authentication strategies was successfully injected
  */
-function triggerPipes(pipes, event, data) {
-  let preparedPipes = [];
-  const wildcardEvent = getWildcardEvent(event);
+function initAuthentication(pluginsManager, plugin) {
+  const
+    mandatoryMethods = ['exists', 'create', 'update', 'delete', 'validate', 'verify'],
+    errorPrefix = `[!] [WARNING][Plugin Manager]: Unable to inject strategies from plugin "${clc.bold(plugin.name)}":`;
+  let valid = true;
 
-  if (pipes && pipes[event] && pipes[event].length) {
-    preparedPipes = pipes[event];
-  }
+  if (!isObject(plugin.object.strategies) || Object.keys(plugin.object.strategies).length === 0) {
+    // eslint-disable-next-line no-console
+    console.warn(clc.yellow(`${errorPrefix} The plugin must provide an object "${clc.bold('strategies')}".`));
 
-  if (wildcardEvent && pipes && pipes[wildcardEvent] && pipes[wildcardEvent].length) {
-    preparedPipes = preparedPipes.concat(pipes[wildcardEvent]);
+    return false;
```

Comment:
should throw a `PluginImplementationError` with the above message, instead of returning `false`